runtime/lua/vim/filetype/detect.lua

```diff
@@ -1322,6 +1322,28 @@ function M.txt(bufnr)
   end
 end
 
+function M.typ(bufnr)
+  if vim.g.filetype_typ then
+    return vim.g.filetype_typ
+  end
+
+  local lines = table.concat(getlines(bufnr, 1, 200))
```

Comment:
That won't work (only when the pattern is at the beginning of the string due to the `^`). Use `ipairs` to iterate over all lines instead (see other detection functions).

Reply:
Indeed ! Done